src/module/iotjs_module_fs.cpp

```diff
@@ -196,42 +197,37 @@ JHANDLER_FUNCTION(Read) {
 
 
 JHANDLER_FUNCTION(Write) {
-  JHANDLER_CHECK(iotjs_jhandler_get_this(jhandler)->IsObject());
-  JHANDLER_CHECK(iotjs_jhandler_get_arg_length(jhandler) >= 5);
-  JHANDLER_CHECK(iotjs_jhandler_get_arg(jhandler, 0)->IsNumber());
-  JHANDLER_CHECK(iotjs_jhandler_get_arg(jhandler, 1)->IsObject());
-  JHANDLER_CHECK(iotjs_jhandler_get_arg(jhandler, 2)->IsNumber());
-  JHANDLER_CHECK(iotjs_jhandler_get_arg(jhandler, 3)->IsNumber());
-  JHANDLER_CHECK(iotjs_jhandler_get_arg(jhandler, 4)->IsNumber());
+  JHANDLER_CHECK_THIS(object);
+  JHANDLER_CHECK_ARGS(5, number, object, number, number, number);
+  JHANDLER_CHECK_ARG_IF_EXIST(5, function);
 
   Environment* env = Environment::GetEnv();
 
-  int fd = iotjs_jhandler_get_arg(jhandler, 0)->GetInt32();
-  int offset = iotjs_jhandler_get_arg(jhandler, 2)->GetInt32();
-  int length = iotjs_jhandler_get_arg(jhandler, 3)->GetInt32();
-  int position = iotjs_jhandler_get_arg(jhandler, 4)->GetInt32();
+  int fd = JHANDLER_GET_ARG(0, number);
+  const iotjs_jval_t* jbuffer = JHANDLER_GET_ARG(1, object);
+  int offset = JHANDLER_GET_ARG(2, number);
+  int length = JHANDLER_GET_ARG(3, number);
+  int position = JHANDLER_GET_ARG(4, number);
+  const iotjs_jval_t* jcallback = JHANDLER_GET_ARG_IF_EXIST(5, function);
 
-  JObject* jbuffer = iotjs_jhandler_get_arg(jhandler, 1);
-  BufferWrap* buffer_wrap = BufferWrap::FromJBuffer(*jbuffer);
+  BufferWrap* buffer_wrap = BufferWrap::FromJBuffer(jbuffer);
   char* data = buffer_wrap->buffer();
   int data_length = buffer_wrap->length();
   JHANDLER_CHECK(data != NULL);
   JHANDLER_CHECK(data_length > 0);
 
   if (offset >= data_length) {
-    JHANDLER_THROW_RETURN(RangeError, "offset out of bound");
+    JHANDLER_THROW(RANGE, "offset out of bound");
   }
```

Comment:
Is it a good idea if the second throw overwrites the first throw? Isn't it create a memory leak?